feat: feature to create 1 PL for several SO irrespective of customers and t…

KrithiRamani opened the pull request:
…o create several DNs grouped by each customer

Documentation Updates:

https://docs.erpnext.com/docs/v13/user/manual/en/stock/delivery-note/edit?wiki_page_patch=eb709e059f
https://docs.erpnext.com/docs/v13/user/manual/en/stock/pick-list/edit?wiki_page_patch=24f2742f11


**Test Cases:**

Customer is not mandatory in PL

Allow selection of multiple SO from several customers

Do not edit Picked Qty in PL. Submit the PL
	- Picked Qty should = Stock Qty

Let qty = 10. Enter 5 as picked qty and submit
	- Picked Qty should remain as 5 and should not be set to stock qty

On submit of PL
	-  “% Picked” (new field) should be updated in SO.
	- “Picked Qty”(new field) should be updated in corresponding SO item row

Create PL for an SO where 
	* “% Picked” is 100%. - On submission, should throw validation message
		* “% Picked” is < 100 % - Should allow submission until 100%

Create PL from multiple SO by clicking on ‘Get Items’
	- submit and create DN from PL 
	- should create separate DNs for each customer.

Create PL by adding items manually(not from SO). Then select an SO also. PL now has both items from SO and those not from SO
	- submit and create DN from PL
	- should create 1 DN for items added manually
	- separate DNs for items from SO, grouped by customer

barredterra committed:
refactor: def instead of lambda

deepeshgarg007 committed:
fix: Grammatical error in comments

deepeshgarg007 committed:
fix: Decide party account debit or credit on payment entry type instead of party type

frappe-pr-bot committed:
fix: Return Qty in PR/DN for legacy data (#27001) (#27003)

(cherry picked from commit 112fc888f1a2ec52d8c736669111fd62e6808109)

Co-authored-by: Marica <maricadsouza221197@gmail.com>

anupamvs committed:
fix: email digest recipient patch

frappe-pr-bot committed:
fix: add child item groups into the filters (#26997) (#27035)

* fix: add child item groups into the filters

* fix: appending values to proper variable

* fix: refactor the loop

(cherry picked from commit c60d5523bca0a0631555a6234a485cd7a1e3c245)

Co-authored-by: Afshan <33727827+AfshanKhan@users.noreply.github.com>

deepeshgarg007 committed:
test: Update test cases

deepeshgarg007 committed:
fix: Revert commit 46372fe

18alantom committed:
feat: process loss report, fix set pl query condition

GangaManoj committed:
fix: Update Product Bundle price based on the rates of its child Items

GangaManoj committed:
fix: Update Product Bundle rate based on its updated amount

GangaManoj committed:
fix: Only update Product Bundle prices if 'Calculate Product Bundle Price based on Child Items' Rates' is enabled

GangaManoj committed:
fix: Return if there are no Items

18alantom committed:
fix: get filters to work
- reorder and rename columns
- add work order filter

frappe-pr-bot committed:
fix: don't allow BOM's item code at any level of child items (#27157) (#27176)

* refactor: bom recursion checking

* fix: dont allow bom recursion

if same item_code is added in child items at any level, it shouldn't be allowed.

* test: add test for bom recursion

* test: fix broken prodplan test using recursive bom

* test: fix recursive bom in tests

(cherry picked from commit c07dce940e31f84fb38ea6fd8d2d41bef619f904)

Co-authored-by: Ankush Menat <ankush@iwebnotes.com>

GangaManoj committed:
fix: Calculate Product Bundle price based on the prices of its child items

GangaManoj committed:
feat: Populate Bundle Items table

GangaManoj committed:
fix: Disable Add Row and Delete Row for Bundle Items

18alantom committed:
fix: pass labels through translation function (#27254)

frappe-pr-bot committed:
Revert "fix: add child item groups into the filters (#26997)" (#27266) (#27269)

This reverts commit c60d5523bca0a0631555a6234a485cd7a1e3c245.

(cherry picked from commit 763450dcf867c31ad954ac5b45ed76e5379f28bf)

Co-authored-by: Afshan <33727827+AfshanKhan@users.noreply.github.com>

frappe-pr-bot committed:
ci(semgrep): add translation checks for report labels (#27280) (#27286)

* ci(semgrep): add translation checks for report labels

* refactor: shift report tests to it's own yml

path can't be applied on a test id basis

* ci: ignore regional report in translation checks

[skip ci]

(cherry picked from commit 72ece75b114524ab2ab4a2b39674b185a21eb9bd)

Co-authored-by: Alan <2.alan.tom@gmail.com>

GangaManoj committed:
fix: Remove redundant description

nextchamp-saqib committed:
fix: undo unintentional changes

frappe-pr-bot committed:
fix(ux): added exception of template item in filters (#27560) (#27675)

(cherry picked from commit 5c249decbb79c5b9436aa7a0f5ca6670993e143e)

Co-authored-by: Noah Jacob <noahjacobkurian@gmail.com>

mergify[bot] committed:
fix(Payment Reconciliation): minor ux fixes (#27779) (#27859)

* fix: minor fixes

* fix: Linters check

* fix: sider check

* fix: kept unallocated payment amount hidden in allocation

* fix: removed Add row button from the tables (redundant)

Co-authored-by: Afshan <33727827+AfshanKhan@users.noreply.github.com>
Co-authored-by: Saqib <nextchamp.saqib@gmail.com>
(cherry picked from commit 5cc3ea0aa71261366436d3e02ab726b89a21a410)

Co-authored-by: Anuja Pawar <60467153+Anuja-pawar@users.noreply.github.com>

GangaManoj committed:
fix: Add tests to validate item

(cherry picked from commit a7ec007dcff77f99d1af9e8f8c473cba629f6efe)

nextchamp-saqib committed:
perf: skip insertion of stock ledger entry

(cherry picked from commit c7fc609236aeaada0375e90b664c041bdbc52570)

ankush committed:
fix: undo changes to allow negative stock flag

(cherry picked from commit 7bafa11d5758c40496898cd0151efde4ff358946)

deepeshgarg007 committed:
fix: TDS round off not working from second transaction

(cherry picked from commit ca0067212dd153566df2965c12aace641ff68720)

GangaManoj committed:
fix: Move test for Finance Books to Depreciation test suite

(cherry picked from commit e8986df3cafa49d1b317429036f4c145ca4f8d86)

pateljannat committed:
fix: conflicts

ankush committed:
fix: remove debug from query

(cherry picked from commit 126ba1674089609adacafaa9a4a380361fdd3a50)

ahmadRagheb committed:
Update selling_settings.json

fix https://github.com/frappe/erpnext/issues/28155

ankush committed:
perf: index for item-sh on repost item valuation

Item-WH based reposting requires querying existing similar repost.
Assuming there is only 1 max extra entry with same params just indexing
item-WH is sufficient to speed up the query.

(cherry picked from commit a5a8c9104fbb5279e6b25c529b6b4c0fc7ccef97)

ankush committed:
fix: default value for allow neg stock in repost_item_valuation

Negative stock can be toggled back after queuing transactions, this
causes failure when repost is executed.

Now allow_negative_stock stock is set at time of queuing the repost job.
This means setting changes done afterwards won't affect already
submitted reposts.

(cherry picked from commit 12e81df2b79a0c7fb7cedb3e467837b60540be04)

marination committed:
test: Purchase Invoice Received Qty

(cherry picked from commit e9c75d6eead9dcdc905d49afec2ed308067d14d5)

marination committed:
fix: (server test) Missing variable declaration

GangaManoj committed:
fix: Fix Product Bundle price calculation when there are multiple Product Bundles

(cherry picked from commit 0803f87660625086f6ea787f85c46f3587463302)

deepeshgarg007 committed:
fix: Taxes and Charges template not getting copied from Purchase Order/Receipt to Invoice

deepeshgarg007 committed:
fix: Better Error logging fordeferred revenue/expense booking

(cherry picked from commit 67a001d87602cb589b5edb606c03be2d2130d594)

ankush committed:
fix: reload sellling settings before patch

rtdany10 committed:
feat: create QR field incase of missing

rtdany10 committed:
fix: disable print format by default

rtdany10 committed:
feat: init KSA POS Invoice

noahjacob committed:
refactor: update_serial_no function and code cleanup

(cherry picked from commit 4f52b86d7e577741f65932601c6ff2066bba1553)

# Conflicts:
#	erpnext/maintenance/doctype/maintenance_visit/maintenance_visit.js

nextchamp-saqib committed:
fix: qty filter not working if apply_multiple_pricing_rules is enabled

(cherry picked from commit 1e17d6a6079d1f65980c3735f180f35fb7d265ab)

nextchamp-saqib committed:
test: add test_multiple_pricing_rules

(cherry picked from commit c0d1f4869f52b4def9d83017577b307e34494e09)

nextchamp-saqib committed:
chore: rename redefinition of multiple pricing rule test

(cherry picked from commit 5a7a9a598b39553901bbd37e8f003943a0446a58)

noahjacob committed:
test: added test for serial no. filters

(cherry picked from commit 9109660a69056424a96bb9ba6d85ccf5214a45c6)

noahjacob committed:
fix(test): fixed test case not creating serials

(cherry picked from commit 90d32006f42a95dba37c51924c5f62be7436dd6f)

noahjacob committed:
refactor: moved purpose table mandatory check to server

(cherry picked from commit b98c61f2a7247d9f27251b8e740664a77a8b51ef)

# Conflicts:
#	erpnext/maintenance/doctype/maintenance_visit/maintenance_visit.js

mergify[bot] committed:
fix: Add Invoice Number field to list view in Opening Invoice Creation Tool (#28941)

ankush committed:
fix: incorrect incoming rate computation for sr no

(cherry picked from commit b9642a1036d7b6fa9a49a08b64145bfc2f56916f)

ankush committed:
fix: get incoming rate only if not return

(cherry picked from commit fcc885bc16403c80f27e6bfb970b156fd2b636b6)

nemesis189 committed:
fix: Allowing non stock items in POS

(cherry picked from commit ec37930404dd1bf4563e4bc2ea4acd767d77aff4)

KrithiRamani committed:
feature to create 1 PL for several SO irrespective of customers and to create several DNs grouped by each customer

KrithiRamani committed:
removed commented code

nemesis189 committed:
fix: sider issues

(cherry picked from commit 6b973d658c9b7f77864627c585908f4c7e7e1920)

nemesis189 committed:
fix: Removed validate_non_stock_items, filter_service_items methods

(cherry picked from commit 9e20fa85c1823bb8622d7321f561b7d010b7d053)

nemesis189 committed:
fix: importing Sum() for qb for test fix

ankush commented:
@KrithiRamani 
- please test for few cases handled using this feature.
- test failure is related
- Lint fix is minor (whitespace); setup pre-commit to automatically do this: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#linting

nemesis189 committed:
fix: linters fix

nemesis189 committed:
fix: linters fix

deepeshgarg007 committed:
fix: Deferred revenue booking for multi currency invoices via JV

(cherry picked from commit 98f294a8ae983fe9bf9b83127fccbf1f1a397ba1)

deepeshgarg007 committed:
fix: Add test case for multicurrency invoice

(cherry picked from commit 094158f287240fd0807ada6eb89dc6874420aed8)

deepeshgarg007 committed:
fix: Test case

(cherry picked from commit e311667a250d112552a115a4b551efdc53cf6099)

ankush committed:
fix(patch): serial no whitespace trimming

old data can contain trailing/leading whitespace which doesn't work well
with code to find last SLE for serial no.

(cherry picked from commit 0faa116f9799f6d921ce8868a8f8eac1756ae008)

mergify[bot] committed:
feat: 'Invoice Number' field in Opening Invoice Creation Tool (#29156)

ruthra-kumar committed:
fix: get project from PO into payment entry

(cherry picked from commit ca17c7226ce343e63cf929023babee425e3da3c5)

ankush committed:
fix: dont update sle values from get_gl_entries

(cherry picked from commit 8f5772463c7f1188110caccae0906d0e98a4b050)

ruthra-kumar committed:
fix: deferred report division by zero exception

(cherry picked from commit fe4d7f86ee7384e794c54b32e5b5f4415654f02a)

deepeshgarg007 committed:
fix: Handle frozen books while handling

(cherry picked from commit 30a647ff8099105a9fdce02bc98d43969985acb3)

deepeshgarg007 committed:
fix: Test case

(cherry picked from commit 18c4ddadf1428133961f3558d7df8eefb7475f40)

KrithiRamani committed:
Merge branch 'frappe:develop' into bulk_creation_of_PL_and_DN

KrithiRamani committed:
added test script for bulk DN creation from PL

mergify[bot] committed:
fix: Task Depends on not removed from Gantt chart (backport #28309) (#29207)

Co-authored-by: conncampbell <conncampbell@conzu.ca>
(cherry picked from commit 31123436fbe223b0116922097108fa444e5092d3)

Co-authored-by: Conn Campbell <conn.campbell@conzu.ca>

deepeshgarg007 committed:
fix(India): NIL Rated, Exempted and non gst invoices in GSTR-1 report

(cherry picked from commit 7d85755595149fb27638126f5d05eb79734fc548)

ruthra-kumar committed:
refactor: get project from basic transactions

- sales order, sales invoice, purchase order and purchase order
- if project not found in transaction, get from items

(cherry picked from commit 8026f86548d0eccc56b1f66e72f47719e7a85969)

ankush committed:
refactor: convert query to ORM

(cherry picked from commit cbaa8fdade4aad306887b23cef9bfeaa17ff07c0)

mergify[bot] committed:
fix: pos invoices consolidation case with permlevel (#29220)

mergify[bot] committed:
fix(gl-report): group by cost center only if include_dimensions is checked (#28883) (#29231)

mergify[bot] committed:
fix(pos): cannot ignore pricing rule for one particular invoice (#29232)

mergify[bot] committed:
fix: "update cost" should ignore overridden routing times #29154 (#29235)

fix: "update cost" should ignore overridden routing times
(cherry picked from commit 754596dfc139a9890b1e446a4d2b2abfeab68449)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: UOM autocomplete broken

All new recent sites seem to have all UOMs as disabled by default. The
desired behaviour is exact opposite of this.

(cherry picked from commit 33aad4b950045d90eb4505274fd8d504776608a8)

ankush committed:
fix(patch): enable all uoms on recently created sites

(cherry picked from commit f8119563ca51f46917cc521429d61a374b6cbc8a)

mergify[bot] committed:
feat: early payment discount on sales & purchase orders (#29238)

mergify[bot] committed:
fix: description not fetched in sales order analytics report (#29254)

mergify[bot] committed:
fix(asset): check if fixed asset account is set against company (#29256)

s-aga-r committed:
fix: consider returned_qty while updating billed_amt

s-aga-r committed:
fix: check so_detail before dn_detail

marination committed:
Merge pull request #29257 from marination/fix-reset-wh-defaults

fix: Avoid resetting Default wh fields for Manufacture Entry
(cherry picked from commit efcfb825d7a72a403f06f62e322cedd7fbbffd29)

dj12djdjs committed:
fix(Payroll): Cannot submit salary slips from amended payroll entry. (#29228)

* fix: salary slip transaction state after payroll entry cancel

* fix: use db_set in on_cancel method

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

marination committed:
fix: Allow multiple fg in repack entry

(cherry picked from commit eac7b5d5aaed83217b0a2843d13e75cb94dbb9a0)

rohitwaghchaure committed:
fix: accounts are coming from different company in the dropdown

(cherry picked from commit dabe5981bbb54a5b2dff704481c4eb3e1676c3fb)

mergify[bot] committed:
fix: filter for leave period in Bulk Leave Policy Assignment (backport #29272) (#29273)

* fix: filter for leave period in Bulk Leave Policy Assignment

* fix: set title for Leave Policy Assignment

(cherry picked from commit 3a18e62a9b9cfa0c84add308f60933d83e74e6bb)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

marination committed:
test: Basic test one item repacked into two

- Also run fg validation and fg marking after checking purpose, avoid unnecessary calls

(cherry picked from commit 3922a39591ad79a44438f3edf28841348a64d74c)

marination committed:
test: Check for FinishedGoodError if 0 FG in repack entry

(cherry picked from commit c49dff385a4798e329f0a8390563afee945c7656)

deepeshgarg007 committed:
fix: Test case

(cherry picked from commit 24bb1e8987c222139884076e4134fe52e3d214f7)

deepeshgarg007 committed:
fix: JSON for nil/exempt and non gst

(cherry picked from commit 55c445cd375a5a16e69205fc702ff86a4577ce48)

deepeshgarg007 committed:
fix: Remove comment

(cherry picked from commit 08f26de3df28eabb93064217f3a67ff9b0ce01b9)

mergify[bot] committed:
fix: threshold fields shows incorrect currency (#29289)

s-aga-r committed:
refactor: use frappe.qb instead of sql

deepeshgarg007 committed:
fix: Remove unwanted test

(cherry picked from commit fd922f1617303d8bf26529b9a9e9d3e919cba011)

mergify[bot] committed:
fix: exclude existing serial numbers while auto creating new #29292 (#29300)

fix: exclude existing serial numbers while auto creating new
(cherry picked from commit 2b681f04fd47b0f73685c8680dcaa7203d671be3)

Co-authored-by: Dany Robert <danyrt@wahni.com>

ankush committed:
test: make sure zero incoming rate is maintained while consuming

(cherry picked from commit 0272397e54513b5a8c18f4e817d3ddf708efc191)

mergify[bot] committed:
fix: only add stock queue if FIFO (#29302) (#29305)

(cherry picked from commit b0cf6195e9efe01150c759483c8ec1c465c746a6)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
test: serial no valuation test case

(cherry picked from commit 5d27a7672e73e514e82fd79b960b16479403e696)

mergify[bot] committed:
fix(patch): sle.serial_no = "\n" causes incorrect queue (#29306) (#29309)

This happens due to old data.

(cherry picked from commit 66bf21f14337a9c83b1ab423a7fafe70781ef21b)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: ignore cancelled SLEs (#29303) (#29307)

(cherry picked from commit 82ea95873049deff46930ea0a792df7609fdf39f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: show work order progress bar even it is closed (#29312) (#29313)

(cherry picked from commit 4d480358289af007618a12e4b74c963d0c1524cb)

Co-authored-by: Anupam Kumar <anupamvns0099@gmail.com>

deepeshgarg007 committed:
fix: Linting issues

(cherry picked from commit 09172002e729a2bbee2362f3132d0cbd688c6b20)

mergify[bot] committed:
feat(UX): Option to exclude holidays while marking monthly attendance (backport #29185) (#29321)

(cherry picked from commit 6aac8de53e5ec57fea3dbdb21d03a6dbc0eade62)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

noahjacob committed:
fix: Serial No. filters for unscheduled visits

(cherry picked from commit dec751377302e504285e834afed6a64dd0f010bf)

nemesis189 committed:
fix: removing `get_value` call by returning is_stock_item in `get_stock_availability` method

(cherry picked from commit ac9a9fb22943a183e9e83077ec428262c4065a15)

nemesis189 committed:
fix: sider fix

(cherry picked from commit 27b35d72e2ba6163c1ea0ba705d018aa7692dc8a)

nemesis189 committed:
fix: remove qty indicator from non stock items

(cherry picked from commit 845c02a989553a8375c94dbd0a8a71687fa9ff07)

mergify[bot] committed:
fix: purchase to Stock UOM conversion on Production Plan #28570 (#29325)

fix: purchase to Stock UOM conversion on Production Plan
(cherry picked from commit 9cd26fbb6a0e5977c35774dc641728beb7b0d655)

Co-authored-by: Maxwell Morais <MaxMorais@users.noreply.github.com>

mergify[bot] committed:
fix: cover case when all material needs to be bought (#29326) (#29327)

When material request is to be made for purchase qty should be converted
to purchase UOM

(cherry picked from commit 48f3d535561efd4d2f3d7b0e38c00d23c68cc517)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: enable allow negative stock by default in reposts (#29331) (#29333)

Negative stock validation is done BEFORE reposting as such it's not
required to be performed again and most likely this fails on old
intermediate data where user has no clue why it failed.

(cherry picked from commit fabe0bce15b3bb1941316072304b244161de3a3e)

Co-authored-by: Ankush Menat <ankush@frappe.io>

barredterra committed:
feat: option to disable Item Tax Template

(cherry picked from commit f60e040d69aa56f71a632b3d77474f60657be689)

barredterra committed:
feat: option to disable tax category

(cherry picked from commit 663c594ead54ced046887bbd94a6a0c88fa7b8e8)

ruthra-kumar committed:
refactor: a more generalized way to clear old entries for test cases

(cherry picked from commit ed0f9cc0639773e1a6880b836832f390556d43a7)

ruthra-kumar committed:
test: unit test for deferred items with zero months

(cherry picked from commit 1e0450ed17fc9e02567ad3a910dc521ad8ce8900)

mergify[bot] committed:
fix: from time and to time not updated in drag and drop action #29114 (#29361)

fix: from time and to time not updated in drag and drop action
(cherry picked from commit 8b5827ed6db1041526b6440ca8e4fde19c646e1e)

Co-authored-by: Mohammed Yusuf Shaikh <49878143+mohammedyusufshaikh@users.noreply.github.com>

deepeshgarg007 committed:
fix: Cleanup empty rows on bank statement import

nextchamp-saqib committed:
chore: undo unnecessary changes

(cherry picked from commit 3da2cac772b0557e15ddf4ee9673381b0d98bca1)

marination committed:
fix: (enhance) BOM Operations Report

- Added filters in the Report for BOM ID, Item Code and Workstation
- Converted Raw SQL to frappe.qb and added method to get filtered data
- Changed fieldtype of 'Time in mins' from Int to Float
- Get BOM wise grouped data to  keep order and accurate grouping in report

(cherry picked from commit a9ff1fc52e1be73fa0d4cbb415607d5d545c545c)

marination committed:
fix: Sider (unused imports) and Linter(add extra empty line)

(cherry picked from commit 3c7b7721123bb2284a00f84f335a8ab1708831f5)

marination committed:
fix: Allow only BOM with ops in BOM ID filter & add total row to report

(cherry picked from commit af734298c8ac541188de4593a064f962147ea27d)

mergify[bot] committed:
fix: show stock UOM for material transfers (#29376) (#29377)

(cherry picked from commit 7e407bcc42260e78ed7275946f71e3e7a89b282f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush commented:
@KrithiRamani the failing test is related, please fix it. 

<img width="1018" alt="Screenshot 2022-01-20 at 3 33 34 PM" src="https://user-images.githubusercontent.com/9079960/150317204-7d4d0ca4-400e-4c84-84ea-c89571f3cd5a.png">

ankush committed:
chore: add translation wrapper

nextchamp-saqib committed:
fix: first depreciation amount in asset

(cherry picked from commit 60bfead773943f4b877483f8cf2b8bbafe6dedba)

nextchamp-saqib committed:
fix: depreciation amount on selling an asset

(cherry picked from commit 0bbde4819bb2d3241b9b9d431cb7b7aa493b1cdd)

mergify[bot] committed:
fix: updated packed_items getting fetched on Sales Return / Credit Note (#28607) (#29387)

* fix: updated packed_items getting fetched on sales return

* fix: updated test cases for product bundle returns

* fix: moved calculate_mapped_packed_items_return to stock.utils

* fix: updated packed_items getting fetched on Return/Credit Note

* refactor: change var name to avoid `self`

* fix: check item code before assigning parent item

(cherry picked from commit 20216fa9f1e98e612c36a89e497de3e92238db7b)

Co-authored-by: Noah Jacob <noahjacobkurian@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
perf: single query to update serial nos

(cherry picked from commit ea5efd0791b1a55b18027c6a07b408ea11fd63cb)

mergify[bot] committed:
fix(Timesheets): calculate to time (backport #28589) (#29390)

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>

ankush committed:
fix: bad default for non-existing serial nos

serial_no field should have empty string if it doesn't have any serial
nos

(cherry picked from commit 33c4a0b5a96178be7d448d15cbe7643f7276cffd)

ankush committed:
fix(patch): update stock reconciliation current serial nos

When current serial nos are non-existing there shouldn't be any value in
current_serial_no field.

(cherry picked from commit 203f0086f8ce3990e47f58095aab40ce49532e74)

ankush committed:
test: create and cancel sr no with stock reco

(cherry picked from commit 4fa93e246eeac8b44da2a850a231c20083352e79)

mergify[bot] committed:
fix: update status on visit cancel (#29389) (#29394)

* fix: update status on cancelling visit

* test: updated test for cancelling visits

(cherry picked from commit 11c9a5a79f2ddb66558658069f10250ac5cbea05)

Co-authored-by: Noah Jacob <noahjacobkurian@gmail.com>

ankush committed:
fix(patch): skip patch if column doesn't exist

deepeshgarg007 committed:
fix: Previous document currency getting overriden

ruthra-kumar committed:
fix: opening invoice creation tool not taking in accouting dimension

(cherry picked from commit e2267cdba85471ce602c90d22ac67e4135a04ad9)

ruthra-kumar committed:
test: validating custom accouting dimension in opening invoice tool

 - create new dimension 'department'
 - assign and validate department in opening invoices

(cherry picked from commit 88463dc494089abcdc7b9655747c75b180b7470d)

Commit:
fix: Don't delete barcode while saving Item (#29411)

mergify[bot] committed:
fix: bypass_credit_limit_check in credit balance report (#29401)

ruthra-kumar committed:
fix: linter issue

(cherry picked from commit 80ea16395336da5dc9b12d561f90fab1eb2e0b00)

ruthra-kumar committed:
refactor: fix failing test cases

(cherry picked from commit f122a47d0db5438cc1fc8b2138153af64a9a3bab)

marination committed:
fix: Use get for conditionally available fields while setting missing values

- Due to custom field "supplier" and missing field "supplier_address", dot operator breaks
- Make sure to use "get" instead of just dot operator if field is in some doctypes, not all

(cherry picked from commit 78b6b29a57928d936181337d2ee5e85d951e5bce)

deepeshgarg007 committed:
fix:Add shipping charges to taxes only if applicable

mergify[bot] committed:
test: clear singles cache for changed settings (#29430) (#29431)

(cherry picked from commit 3d3f0139fd2df42a7753c7d2560e2451819dee92)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
refactor: reusable clean_serial_nos

(cherry picked from commit b20df3745eafdcc18d3b7cb9a93065562aef5c6f)

mergify[bot] committed:
fix: honour 'include holidays' setting while marking attendance for leave application (backport #29425) (#29432)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

ankush committed:
fix: extend sr_no cleanup to packed items too

(cherry picked from commit e177c5277f223cd0f513d7189053e444554bd745)

ankush committed:
fix: work order serial no allows storing whitespace

(cherry picked from commit 43bd88e741b69fcfd4a2547fd59e5e6286229522)

mergify[bot] committed:
fix: fetch "transfer material against" from BOM (#29435) (#29436)

(cherry picked from commit b8fbfed096a4e5f0e1ac97fd7a35a49b0a316b62)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: remove all stock UOM conversion when changing stock UOM (#29463)

(cherry picked from commit eb8d08b411b66797d9f3892e079c49828335d34a)

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
fix: Further sort sales_order_analysis to get consistent response

Since the data was sorted by `transaction_date`, report sorting order used to be inconsistent everytime if there were lots of records with same transaction date. This used to create problems while exporting report as "Excel" with filters.

(cherry picked from commit 29c7947db05106cb87a0dfe0905dc3abc8c63cd4)

mergify[bot] committed:
fix: stock ledger rendering glitch (#29482)

undefined Issue

(cherry picked from commit f38004219518e901eb736780b181c19fe88ce823)

Co-authored-by: smehata <53169014+smehata@users.noreply.github.com>
(cherry picked from commit 0dafa8a12a0c0745bbabceb457332075fd6db9f9)

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

ankush committed:
fix: ensure correct-ish transfer against

(cherry picked from commit 86efa0dbfd45f21d72d047ef77ca0cb8ae2b8d94)

ankush committed:
chore: disable annoying flake8 formatting for QB (#29488)

[skip ci]

(cherry picked from commit 251576a6bb38c9b0f0793330e10e256fe7898e5c)

ankush committed:
test: delete data after running test

[skip ci]

ankush committed:
fix: ignore alternate item while checking pending qty

(cherry picked from commit 14e3e163aeec88a61748beca415e59bc0aa62870)

deepeshgarg007 committed:
fix(India) Tax calculation for overseas suppliers

(cherry picked from commit e45c38337c3afc78347f0d26ecd8365148b26d83)

ankush committed:
fix: add unique constraint on bin at db level

(cherry picked from commit 08810dbcce3c96855c2c5bfe266ba9d7ea61fc42)

ankush committed:
fix: ignore None item/wh whie updating reservation

(cherry picked from commit a1e7771cdd7713c64ab525ba7955a139712c727e)

s-aga-r committed:
fix(patch): patch duplicate bins

(cherry picked from commit 7ff3ca25e5b8916d9ebb375f7574dc1a5ebdd4ff)

ankush committed:
refactor: patch for fixing broken bins

fix(patch): delete fully broken bins

if bin doesn't have item_code or warehouse then it's not recoverable.

(cherry picked from commit c2ecc7a2d1da839423fd768821b1f77ddcf7f53d)

mergify[bot] committed:
fix: cost of poor quality report time filters not working (#28958) (#29521)

* fix: cost of poor quality report time filters not working

* chore:update cost of poor quality report to use query builder

* fix: linter warnings

* chore: updated report query

* chore: added test filters

* fix : cleared linter warnings

* chore: formatting

* refactor: query generation

- optionally apply date filters
- join instead of expensive sub-query
- return as dictionary

* test: simplify test

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 0f7c2a19de34c9bd14d457c9b6e7f6cf279c6ea2)

Co-authored-by: aaronmenezes <ron2805@gmail.com>

nemesis189 committed:
fix: Validation for invalid serial nos at POS invoice level (#29447)

(cherry picked from commit 05bbb69d0ed53e6687692b72f046d6bf07a4eb08)

ankush committed:
fix: update reserved qty for production/ s/c

(cherry picked from commit 0a1533446495f27445afbcb9d58c30d4d31e7b20)

mergify[bot] committed:
fix: do not hide Loan Repayment Entry field in salary slip (backport #29535) (#29551)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

nextchamp-saqib committed:
test: point of sale search

(cherry picked from commit 650d44a714cb94c9afdd80d1c38ce9c1e58cf28c)

nextchamp-saqib committed:
chore: remove useless rollback

(cherry picked from commit 2aca54eb383c4e99861e40a6d1d8524ffd66f2ae)

nextchamp-saqib committed:
chore: remove unused import

(cherry picked from commit 4e4159ec0668ddecb9684e3c96810b5fca10b29a)

ruthra-kumar committed:
fix: typeerror on invoice creation from SO/PO

(cherry picked from commit 9bd56b0f79af4970ce6c1762d647725fba4ebbf9)

nextchamp-saqib committed:
fix: flaky test

(cherry picked from commit bf70feb7c9be7320188d72e26b2ef5d212c7a290)

mergify[bot] committed:
fix: leave application tests (backport #29539) (#29563)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

mergify[bot] committed:
fix: don't show "create" on cancelled BOMs (#29570) (#29572)

(cherry picked from commit a3a05c0c23be03edf46353ee3121426485eca36f)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: employee reminders fixes (backport #29548) (#29576)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

s-aga-r committed:
fix: ordered_qty for production-plan-item

mergify[bot] committed:
fix: Program Enrollment tests (backport #29592) (#29594)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

nextchamp-saqib committed:
fix(pos): pricing rule on transactions doesn't work

(cherry picked from commit e082e3f702a4bc8993411115504baa3e3ff1d996)

nemesis189 committed:
fix: Coupon code item pricing dynamic updation issue

(cherry picked from commit ccf63124d62139c586a3d6737460a67a942956b1)

deepeshgarg007 committed:
fix: Incorrect provisional profit and loss in balance sheet

(cherry picked from commit ed2c6b6637960fe67846f933049d288078993b92)

deepeshgarg007 committed:
fix: Zero rated exports in GSTR-3B report

(cherry picked from commit 233e6449fc7fe1d09b1a034bb3491aca0b39028d)

mergify[bot] committed:
fix: ignore empty customer/supplier in item query (#29610) (#29613)

* fix: dont try to filter by customer/supplier if None

* test: item query with emtpy supplier

(cherry picked from commit 41a95e56241ff8f3dceac7285f0bc6b9a43d7a06)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: add default if qty is not found

marination committed:
test: Packed Items test file

(cherry picked from commit 4e7b4dc9a845c8090e5c97010ce857caafb000dd)

marination committed:
fix: (Linter) import sequence

(cherry picked from commit f18ec2d94705022fe6fd5ae87112f7e59be21651)

mergify[bot] committed:
feat: show stock value difference in stock ledger report (#29607) (#29621)

(cherry picked from commit 04f6426dbf47a14b4a2950f2b3d373dd40d83347)

Co-authored-by: Ankush Menat <ankush@frappe.io>

deepeshgarg007 committed:
fix: Incorrect tax template in Sales Invocie via data import

deepeshgarg007 committed:
fix: Ignore linked invoices on Journal Entry cancel

(cherry picked from commit 6f7ae6290779181d1ad6db11f716a50f8c94072a)

mergify[bot] committed:
fix: enable Allow on Submit for 'Is Active' field in Salary Structure (backport #29630) (#29643)

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

ruchamahabal committed:
test: earned leave allocation for passed months and allocation on month-end

(cherry picked from commit 63ee4f1b64b0110d6d97f4114605db2732dcb224)

ankush committed:
fix: consider packed items too when reposting

(cherry picked from commit c56d07dee3f0c35e211424d82fec628aea22d2e9)

ankush committed:
fix: ignore cancelled svd while updating GLE for PR

This happens because LCV cancels and reposts entries so unless filtered
by non-cancelled entries you can randomly get old values.

(cherry picked from commit 8858c703a894ec35f5e44a91b82a7cc1f40ba2b4)

ankush committed:
test: regression test for LCV GL entries

(cherry picked from commit 69c65afd729d87115ee469f2bf8240af43d55e65)

deepeshgarg007 committed:
fix: Billing status for zero amount ref doc

(cherry picked from commit d6796924728e998c3331e3af8c58556b0bbd7db1)

deepeshgarg007 committed:
fix: Add test case

(cherry picked from commit c5726fd3da410874c4532887d2c625c54bf6de6a)

ankush committed:
refactor: simplify get_items_and_warehouses

Also remove dead code related to stock reconciliation_json.

(cherry picked from commit e6ab8df8f2c48932a7368c5ac69ebcac14cf015c)

ankush committed:
fix: merge stock ledger item warehouse with doc's

(cherry picked from commit 1022db04745b3c6f16710b43720fef4f0fd0d295)

ankush committed:
chore: drop dead field from stock reconciliation

(cherry picked from commit 43f8ee1dd1525b6cf3e88154bec314aca7b23ca5)

ankush committed:
test: commit item/warehouse creation to db

(cherry picked from commit f82d7eb73fe6dd70a98fa6656e90ed9c6565be16)

abhinavxd committed:
fix: missing key in loan

(cherry picked from commit d0043bdbace8cac51527c9998435aa5c4f438b25)

ankush committed:
fix: dont show cancelled PO items in plan report

(cherry picked from commit 6459ceaea14cc8422de83dbca060e1c69b0e1d13)

mergify[bot] committed:
fix(ux): make stock entry type the title field (#29674) (#29675)

(cherry picked from commit 5e6227e3d87cc54e842cb6f71f18083055c9a51a)

Co-authored-by: Ankush Menat <ankush@frappe.io>

deepeshgarg007 committed:
fix: Add disbursement accounts to tests

(cherry picked from commit 8ece2845f2dd6cc382c21c7794ee9d533cf4ea9b)

deepeshgarg007 committed:
test: Add test case for repayment against partially disbursed loans

(cherry picked from commit ef69d1fd385bfe740ce1765f1a4998d27456ce79)

phot0n committed:
fix: use item_code instead of parent field in bom_stock_calculated report (#29684)

(cherry picked from commit 72a812f18bfd27842156d7b1afb1f301fbead7ed)

nextchamp-saqib committed:
chore: show credit/debit-to account in error message

(cherry picked from commit e93bb8a3364c04c8a30b47f681e53747140e4fd9)

mergify[bot] committed:
fix: ignore cancelled SLEs (#29679) (#29698)

(cherry picked from commit 0ca60afc3fb7190cbba58ef42b84c51bffb9d660)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: dont ignore items that dont have SVD (backport #29656) (#29694)

* fix: dont ignore items that dont have SVD

When items go from negative to positive stock value diff can be zero but
item might have taxes / need divisional loss adjustment.

(cherry picked from commit c88c368880a134b12ad82d7674cb5e12d5a858ba)

* test: check when PR moves stock from neg to pos

(cherry picked from commit d1f57538855eaa8c6598a553a8b776b3ab511171)

Co-authored-by: Ankush Menat <ankush@frappe.io>

bhavesh95863 committed:
fix: ignore rate validation for work order (#29690)

* fix: ignore rate validation for work order

* test: validate on save instead of on creation

Co-authored-by: Ankush Menat <ankush@frappe.io>

nextchamp-saqib committed:
fix: currency in bank reconciliation chart

(cherry picked from commit 0fc5d2278d2785385051d576e895c8286ad7a3a2)

mergify[bot] committed:
fix: fetch image form item (backport #29523) (#29571)

* fix: fetch image form item (#29523)

(cherry picked from commit 319322228aeb39e2d76c8b7bfd23a335f725579a)

# Conflicts:
#	erpnext/assets/doctype/asset/asset.json

* chore: resolve merge conflicts

Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>
Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

KrithiRamani committed:
corrected test data

KrithiRamani commented:
Too many issues coz of rebase. Opened another PR from latest develop
